### PR TITLE
JUCX: add memory type hint to ucp operations.

### PR DESCRIPTION
**Pull request by petro-rudenko**
## What
add memory type hint to all ucp operations.

## Why ?
Would be implemented in NBX API (from #6067). This is backward compatible since by default it passes `UCS_MEMORY_TYPE_UNKNOWN`

### CODE REVIEW

File: bindings/java/src/main/java/org/openucx/jucx/ucp/UcpWorker.java [hunk 1]
```diff
@@ -222,9 +245,22 @@ public UcpRequest recvTaggedMessageNonBlocking(ByteBuffer buffer, UcpTagMessage
...
 
+    public void amDataRelease(long address) {
+        amDataReleaseNative(getNativeId(), address);
+    }
+
+    public UcpRequest recvAmDataNonBlocking(long dataDesc, long address, long size,
+                                            UcxCallback callback, int memoryType) {
+        return recvAmDataNonBlockingNative(getNativeId(), dataDesc, address, size, callback,
+            memoryType);
+    }
```
Comment (yosefe) [inline bindings/java/src/main/java/org/openucx/jucx/ucp/UcpWorker.java:262]: unrelated?

File: bindings/java/src/main/java/org/openucx/jucx/ucp/UcpWorker.java [hunk 2]
```diff
@@ -261,20 +297,30 @@ public ByteBuffer getAddress() {
...
 
     private static native UcpRequest recvTaggedIovNonBlockingNative(long workerId,
                                                                     long[] localAddresses,
                                                                     long[] sizes,
                                                                     long tag, long tagMask,
-                                                                    UcxCallback callback);
+                                                                    UcxCallback callback,
+                                                                    int memoryType);
+
+    private static native UcpRequest recvAmDataNonBlockingNative(long workerId, long dataDesc,
```
Comment (yosefe) [inline bindings/java/src/main/java/org/openucx/jucx/ucp/UcpWorker.java:310]: seems from another PR

**Review by yosefe — COMMENTED** [review]
maybe worth adding implementation and test to this PR?

**Comment by petro-rudenko** [comment]
The actual passing this parameter to ucp would be in another PR when switching to NBX. Trying to split it, to smaller PRs. All the existing tests should pass with it, since it passes `UCS_MEMORY_TYPE_UNKNOWN` everywhere.

**Review by yosefe — APPROVED** [review]

**Comment by yosefe** [comment]
bot:pipe:retest

**Comment by yosefe** [comment]
bot:pipe:retest
